runtime/include/cloe/databroker.hpp

```diff
@@ -0,0 +1,1254 @@
+    // Setter-function types
+    template <typename TValue>
+    using typed_set_value_function_t = std::function<void(const TValue*)>;
+    using type_erased_set_value_function_t = std::any;
+    // Event types
+    template <typename TValue>
+    using typed_value_changed_event_t = Event<const TValue*>;
+    using type_erased_value_changed_event_t = std::any;    
+    // Event trigger-function types
+    template<typename TValue>
+    using typed_on_value_change_event_function_t = std::function<void(const TValue*)>;
+    using type_erased_on_value_change_event_function_t = std::any;
+
+    /**
+     * Name of the signal
+     */
+    std::vector<std::string> names_{};
+    /**
+     * std::type_info of the signal
+     */
+    const std::type_info* type_ {nullptr};
+    /**
+     * getter-function
+     */
+    type_erased_get_value_function_t get_value_{};
+    /**
+     * setter-function
+     */
+    type_erased_set_value_function_t set_value_{};
+    /**
+     * Event which gets raised when the signal value changes
+     */
+    type_erased_value_changed_event_t value_changed_event_{};
+    /**
+     * Triggers the value-changed event
+     */
+    type_erased_on_value_change_event_function_t on_value_changed_{};
+    /**
+     * std::function returning the count of event subscribers
+     */
+    std::function<std::size_t()> subscriber_count_{};
+
+    /**
+     * @brief Private default c'tor
+     * @note: Implementation-#2: The class shall be created only via a factory-method
+     */
+    Signal() = default;
+public:
+    /**
+     * @brief Public c'tor, accessible only via private access-token
+     * @note: Design-#1: The class shall be constructible via external helpers
+     */
+    explicit Signal(access_token):Signal(){}
+    Signal(const Signal&) = delete;
+    Signal(Signal&&) = default;
+    virtual ~Signal() = default;
+    Signal& operator =(const Signal&) = delete;
+    Signal& operator =(Signal&&) = default;
+
+private:
+    /**
+     * @brief Validates that the stored signal type matches the template type
+     * @tparam TValue Expected type of the signal
+     */
+    template<typename TValue>
+    void validate_type() const
+    {
+        const std::type_info* requested_type = &typeid(TValue);
+        if ((type_ == nullptr) || (requested_type == nullptr) || (type_ != requested_type))
+        {
+            throw std::logic_error("Type mismatch between signal and requested type");
+        }
+    }
+
+public:
+    /**
+     * @brief Returns the getter function of the signal
+     * @tparam TValue Type of the signal
+     * @return Getter function of the signal
+     */
+    template<typename TValue>
+    typename enable_if_is_wanted<TValue, const typed_get_value_function_t<TValue>*>::type getter() const
+    {
+        validate_type<TValue>();
+        const typed_get_value_function_t<TValue>* get_value_fn = std::any_cast<typed_get_value_function_t<TValue>>(&get_value_);
+        return get_value_fn;
+    }

```

Comment:
Can we elide both definitions and do something like this at compile time?

```suggestion
    {
        if constexpr (is_unwanted) {
           compiler_trap;
        }
        validate_type<TValue>();
        const typed_get_value_function_t<TValue>* get_value_fn = std::any_cast<typed_get_value_function_t<TValue>>(&get_value_);
        return get_value_fn;
    }

```

Alternatively, can we move all compiler trap declarations to a separate file (that I can ignore as a user when looking at the header.)

Reply:
The suggestion triggers for any instantiation a compiler error.

As outlined, the reason is that the static-expression does not prevent the instantiation of the encapsulated staticexpression which (intentionally) raises the compiler error. A compile-time if-then-else has to be implemented via specialization or SFINAE.